test/integration/singlecluster/controller/jobs/job/job_controller_test.go

```diff
@@ -2667,3 +2668,325 @@ var _ = ginkgo.Describe("Job controller when TopologyAwareScheduling enabled", g
+					AfterDeactivatedByKueue: afterDeactivatedByKueue,
+				},
+			},
+			WaitForPodsReady: waitForPodsReady,
+		}
+
+		fwk.StartManager(ctx, cfg, managerAndControllersSetup(false, false, configuration,
+			jobframework.WithObjectRetentionPolicies(configuration.ObjectRetentionPolicies),
+			jobframework.WithWaitForPodsReady(waitForPodsReady),
+		))
+
+		ns = util.CreateNamespaceFromPrefixWithLog(ctx, k8sClient, "core-")
+
+		fl = testing.MakeResourceFlavor("fl").Obj()
+		util.MustCreate(ctx, k8sClient, fl)
+
+		cq = testing.MakeClusterQueue("cq").
+			ResourceGroup(*testing.MakeFlavorQuotas(fl.Name).Resource(corev1.ResourceCPU, "5").Obj()).
+			Obj()
+		util.MustCreate(ctx, k8sClient, cq)
+
+		lq = testing.MakeLocalQueue("lq", ns.Name).ClusterQueue(cq.Name).Obj()
+		util.MustCreate(ctx, k8sClient, lq)
+	})
+
+	ginkgo.JustAfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, cq, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, fl, true)
+
+		fwk.StopManager(ctx)
+	})
+
+	ginkgo.When("WaitForPodsReady disabled", func() {
+		ginkgo.When("tiny deactivation retention period", func() {
+			ginkgo.BeforeEach(func() {
+				enableWaitForPodsReady = true
+				afterDeactivatedByKueue = &metav1.Duration{Duration: util.TinyTimeout}
+			})
+
+			ginkgo.It("shouldn't delete job when it is deactivated manually", func() {
+				job := testingjob.MakeJob("job", ns.Name).
+					Queue(kueue.LocalQueueName(lq.Name)).
+					Request(corev1.ResourceCPU, "2").
+					Obj()
+				ginkgo.By("Creating a Job", func() {
+					util.MustCreate(ctx, k8sClient, job)
+				})
+
+				wlKey := types.NamespacedName{
+					Namespace: job.Namespace,
+					Name:      workloadjob.GetWorkloadNameForJob(job.Name, job.UID),
+				}
+				wl := &kueue.Workload{}
+
+				ginkgo.By("Waiting for the Workload to be created", func() {
+					gomega.Eventually(func(g gomega.Gomega) {
+						g.Expect(k8sClient.Get(ctx, wlKey, wl)).To(gomega.Succeed())
+					}, util.Timeout, util.Interval).Should(gomega.Succeed())
+				})
+
+				ginkgo.By("Admitting the Workload", func() {
+					admission := testing.MakeAdmission(cq.Name).
+						Assignment(corev1.ResourceCPU, kueue.ResourceFlavorReference(fl.Name), "1m").
+						AssignmentPodCount(wl.Spec.PodSets[0].Count).
+						Obj()
+					gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, wl, admission)).Should(gomega.Succeed())
+					util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, wl)
+				})
+
+				ginkgo.By("Deactivating the Workload", func() {
+					util.DeactivateWorkload(ctx, k8sClient, wlKey)
+				})
+
+				ginkgo.By("Checking that the Job is deleted", func() {
+					createdJob := &batchv1.Job{}
+					gomega.Eventually(func(g gomega.Gomega) {
+						err := k8sClient.Get(ctx, client.ObjectKeyFromObject(job), createdJob)
+						if apierrors.IsNotFound(err) {
+							return
+						}
+						g.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
```suggestion
						g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(job), createdJob)).To(utiltesting.BeNotFoundError)
```

Could you use a more proper way?